x-pack/plugins/embeddable_enhanced/public/plugin.ts

```diff
@@ -61,11 +64,21 @@ export class EmbeddableEnhancedPlugin
 
   public setup(core: CoreSetup<StartDependencies>, plugins: SetupDependencies): SetupContract {
     this.setCustomEmbeddableFactoryProvider(plugins);
-
+    const startServices = createStartServicesGetter(core.getStartServices);
     const panelNotificationAction = new PanelNotificationsAction();
     plugins.uiActionsEnhanced.registerAction(panelNotificationAction);
     plugins.uiActionsEnhanced.attachAction(PANEL_NOTIFICATION_TRIGGER, panelNotificationAction.id);
 
+    plugins.uiActionsEnhanced.registerDrilldown(
+      new UrlDrilldown({
+        getGlobalScope: urlDrilldownGlobalScopeProvider({ core }),
+        navigateToUrl: (url: string) =>
+          core.getStartServices().then(([{ application }]) => application.navigateToUrl(url)),
+        getOpenModal: () => core.getStartServices().then(([{ overlays }]) => overlays.openModal),
+        getSyntaxHelpDocsLink: () => startServices().core.docLinks.links.dashboard.drilldowns, // TODO: replace with docs https://github.com/elastic/kibana/issues/69414
```

Comment:
It seems two different ways to access start services are used here: (1) async `core.getStartServices()` and (2) sync `startServices`. Let's use one way for consistency.

Reply:
I'd use 1 method (`async`), but in case of `syntaxDocs` using `async` is very inconvenient, so I am using `sync` for that single one as an escape hatch.